openwisp_users/admin.py

```diff
@@ -58,6 +58,14 @@ def has_add_permission(self, *args, **kwargs):
         """
         return False
 
+    def has_change_permission(self, request, obj=None):
+        if obj and request.user.pk != obj.pk:
```

Comment:
If the user is superuser, things can work as before without the need of doing any query